C/services/common/include/binary_plugin_handle.h

```diff
@@ -24,14 +24,26 @@ class BinaryPluginHandle : public PluginHandle
 	public:
 		// for the Storage plugin
 		BinaryPluginHandle(const char *name, const char *path, tPluginType type) {
+			dlerror();	// Clear the existing error
 			handle = dlopen(path, RTLD_LAZY);
+			if (!handle)
+			{
+				Logger::getLogger()->error("Unable to load storage plugin %s, %s",
```

Comment:
Here, it can be non-storage plugin also?

Reply:
Really! Then the comments you added when you wrote the code are incorrect in that case!

Reply:
My mistake, I think I misread this log statement to be in get_plugin_info.cpp. Please ignore my comment above.